lua/plenary/scandir.lua

```diff
@@ -0,0 +1,318 @@
+local Path = require'plenary.path'
+local os_sep = Path.path.sep
+
+local uv = vim.loop
+
+local split = function(str, pat)
+  local concat_pat = '[^' .. pat .. ']+'
+  local res = {}
+  for s in str:gmatch(concat_pat) do
+    table.insert(res, s)
+  end
+  return res
+end
```

Comment:
We need a own split because i was using `vim.fn.split` in a uv callback.
I am not sure if thats correct but it works here 🤔